02_classify_compressed/nbconverted/plot_mutation_compressed_results.py

```diff
@@ -0,0 +1,303 @@
+
+
+# In[5]:
+
+
+def label_points(x, y, gene, sig, ax):
+    text_labels = []
+    a = pd.DataFrame({'x': x, 'y': y, 'gene': gene, 'sig': sig})
+    for i, point in a.iterrows():
+        if point['y'] > -np.log10(SIG_ALPHA):
+            text_labels.append(
+                ax.text(point['x'], point['y'], str(point['gene']))
+            )
+    return text_labels
+
+compare_df = (
+    compare_df.rename(columns={'identifier': 'gene'})
+              .sort_values(by=['n_dims', 'training_data'])
+)
+compare_df['nlog10_p'] = -np.log10(compare_df.corr_pval)
+
+sns.set({'figure.figsize': (24, 18)})
+sns.set_style('whitegrid')
+fig, axarr = plt.subplots(3, 3)
+
+# plot mutation prediction from various data types, in a volcano-like plot
+for row_ix, n_dims in enumerate(compare_df.n_dims.unique()):
+    for col_ix, training_data_type in enumerate(compare_df.training_data.unique()):
+        ax = axarr[row_ix, col_ix]
+        data_df = compare_df[
+            (compare_df.training_data == training_data_type) &
+            (compare_df.n_dims == n_dims)
+        ]
+        sns.scatterplot(data=data_df, x='delta_mean', y='nlog10_p', hue='reject_null',
+                        hue_order=[False, True], ax=ax)
+        # add vertical line at 0
+        ax.axvline(x=0, linestyle=':', color='black')
+        # add horizontal line at statistical significance threshold
+        l = ax.axhline(y=-np.log10(SIG_ALPHA), linestyle=':')
+        # label horizontal line with significance threshold
+        # (matplotlib makes this fairly difficult, sadly)
+        ax.text(0.9, -np.log10(SIG_ALPHA)+0.2,
+                       r'$\alpha = {}$'.format(SIG_ALPHA),
+                       va='center', ha='center', color=l.get_color(),
+                       backgroundcolor=ax.get_facecolor())
+        ax.set_xlabel('AUPR(signal) - AUPR(shuffled)')
+        ax.set_ylabel(r'$-\log_{10}($adjusted $p$-value$)$')
+        ax.set_xlim((-0.2, 1.0))
+        y_max = max(compare_df.nlog10_p)
+        ax.set_ylim((0, y_max+5))
+        ax.legend(title=r'Reject $H_0$', loc='upper left')
+        ax.set_title(r'{} PCs, {} data ({}/{} significant genes)'.format(
+            n_dims, training_data_type,
+            np.count_nonzero(data_df.reject_null),
+            data_df.shape[0]
+        ))
+
+        # label genes and adjust text to not overlap
+        # automatic alignment isn't perfect, can align by hand in inkscape if necessary
+        text_labels = label_points(data_df['delta_mean'],
+                                   data_df['nlog10_p'],
+                                   data_df.gene,
+                                   data_df.reject_null,
+                                   ax)
+        adjust_text(text_labels,
+                    ax=ax, 
+                    expand_text=(1., 1.),
+                    lim=5)
+
+
+# In[6]:
+
+
+sns.set({'figure.figsize': (18, 6)})
+fig, axarr = plt.subplots(1, 2)
+
+# plot mean performance over all genes in Vogelstein dataset
```

Comment:
When you're farther along, might be helpful to pull this into a function since I see you're making similar plots in other ntoebooks

Reply:
Good idea! I'll create an issue